src/lib/merkletree/merkletree.ts

```diff
@@ -61,7 +61,7 @@ export class Merkletree {
     const newRootKey = await this.addLeaf(newNodeLeaf, this.root, 0, path);
     this.#root = newRootKey;
 
-    await this.#db.setRoot(this.root);
+    this.#db.setRoot(this.root);
```

Comment:
maybe let's do this as await, someone can implement custom storage that is async